web/src/filter.ts

```diff
@@ -879,6 +890,8 @@ export class Filter {
         this.cached_sorted_terms_for_comparison = undefined;
         if (this.has_operator("channel")) {
             this._sub = stream_data.get_sub_by_name(this.operands("channel")[0]!);
+        } else {
+            this._sub = undefined;
```

Comment:
This is for case when `#narrow/channel/topic/with/private_msg` -- when the narrow terms are adjusted and new filter with only dm terms is set up, the `_sub` of old filter needs to be removed.